### PR TITLE
use a scheduled function for settimeofday_cb

### PR DESCRIPTION
**Pull request by d-a-v**
fixes https://gitter.im/esp8266/Arduino?at=5d977042940b4c2fc074b949

### CODE REVIEW

File: cores/esp8266/sntp-lwip2.cpp [hunk 1]
```diff
@@ -42,16 +42,22 @@
 #include <osapi.h>
 #include <os_type.h>
 #include "coredecls.h"
+#include "Schedule.h"
 
-extern "C" {
+std::function<void()> _settimeofday_cb;
```
Comment (devyte) [inline cores/esp8266/sntp-lwip2.cpp]: Please use anonymous namespace or static

File: cores/esp8266/coredecls.h [hunk 2]
```diff
@@ -17,14 +17,18 @@ extern bool timeshift64_is_set;
...
 void disable_extra4k_at_link_time (void) __attribute__((noinline));
 
 uint32_t sqrt32 (uint32_t n);
 uint32_t crc32 (const void* data, size_t length, uint32_t crc = 0xffffffff);
 
 #ifdef __cplusplus
 }
+
+#include <functional>
+void settimeofday_cb (std::function<void()>&& cb);
```
Comment (devyte) [inline cores/esp8266/coredecls.h]: Considering the number of repetitions I see, I think the arg type here should get a type alias, e.g.:
```cpp
using timeOfDayCB = std::function<void()>;
void setTimeOfDayCB (timeOfDayCB &&cb);
```

Comment (devyte) [reply cores/esp8266/coredecls.h]: Also, please camelCase.

Comment (d-a-v) [reply cores/esp8266/coredecls.h]: `settimeofday()` is a libc function.
Using `setTimeOfDayCB()` makes it quite unrelated with the system function.

File: cores/esp8266/coredecls.h [hunk 3]
```diff
@@ -17,14 +17,21 @@ extern bool timeshift64_is_set;
...
 
 uint32_t sqrt32 (uint32_t n);
 uint32_t crc32 (const void* data, size_t length, uint32_t crc = 0xffffffff);
 
 #ifdef __cplusplus
 }
+
+#include <functional>
+
+using functionalVoidVoid = std::function<void()>;
```
Comment (devyte) [inline cores/esp8266/coredecls.h]: This is rather confusing. 
How about just functionalVoid? Alternatively, a function that takes no args is called a nullary function. A function that doesn't return a value is called a void function. How about:
-nullaryVoid
-functionalNullaryVoid
-functionalVoidNullary
-some variation of the above

**Pull request merged**